bdo element uses bidi-override instead of isolate-override

Ahmad-S792 opened the pull request:
#### 1bf83c07887e3a653a605cf08600e1774247a3c9
<pre>
bdo element uses bidi-override instead of isolate-override

<a href="https://bugs.webkit.org/show_bug.cgi?id=283765">https://bugs.webkit.org/show_bug.cgi?id=283765</a>
<a href="https://rdar.apple.com/problem/140662417">rdar://problem/140662417</a>

Reviewed by Alan Baradlay.

This patch aligns WebKit with web specification [1]:

<a href="https://html.spec.whatwg.org/multipage/rendering.html#bidi-rendering">https://html.spec.whatwg.org/multipage/rendering.html#bidi-rendering</a>

As per web specification, `bdo` should be `unicode-bidi: isolate-override`
instead of `bidi-override`. This also align us with Gecko / Firefox and
Blink / Chromium.

It is covered by existing imported WPT test case below:
&gt; html/rendering/bidi-rendering/unicode-bidi-ua-rules.html

Since we haven&apos;t aligned with web specification fully, we progress only few more
sub-tests pass for `UA stylesheet rule for unicode-bidi, for &lt;bdo&gt;`.

In 259172@main , WebKit fixed bug to handle bidirectional override and corresponding
glyph content width but it failed to account for `isolate-override`, this fixes it as well.

* Source/WebCore/css/html.css:
(bdo, bdo[dir]):
(bdo): Deleted.
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(TextUtil::width):
* LayoutTests/fast/css/default-bidi-css-rules-expected.txt:
* LayoutTests/fast/css/default-bidi-css-rules.html:
* LayoutTests/fast/text/international/bidi-LDB-2-HTML.html:

Canonical link: <a href="https://commits.webkit.org/287175@main">https://commits.webkit.org/287175@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32f7afed1e84034cf4539367e2750fd57ac55215

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78646 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57691 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32028 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83307 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29908 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/80779 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66842 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5972 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61597 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19516 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81713 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51624 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/70342 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41906 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48970 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/25538 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28249 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70074 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25915 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84675 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6011 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4136 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69824 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6172 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67597 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69078 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17205 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13109 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/11622 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5958 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/11889 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5944 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9380 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7733 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on LayoutTests/fast/text/international/bdo-bidi-width.html:
```diff
@@ -1,4 +1,5 @@
 <!DOCTYPE html>
+<meta name="fuzzy" content="maxDifference=192; totalPixels=286-369" />
```
this looks like a lot!

Ahmad-S792 replied:
@alanbaradlay - found the fix. :-) This test was added in [259172@main](https://commits.webkit.org/259172@main) and it was based on the [Blink's commit](https://chromium.googlesource.com/chromium/blink/+/fe79ee4457111490064d8bf9b6c5dce97f26b5d4). Although the blink commit used `isOverride`, which covers both `override` and `isolate-override` but our patch was only for `override`, so when I manually updated locally to have `OR` between then it works similar to Safari 18.2. So I will adjust it in this as well. :-)

Ahmad-S792 replied:
Fixed. :-)

Ahmad-S792 committed:
bdo element uses bidi-override instead of isolate-override

https://bugs.webkit.org/show_bug.cgi?id=283765
rdar://problem/140662417

Reviewed by Alan Baradlay.

This patch aligns WebKit with web specification [1]:

https://html.spec.whatwg.org/multipage/rendering.html#bidi-rendering

As per web specification, `bdo` should be `unicode-bidi: isolate-override`
instead of `bidi-override`. This also align us with Gecko / Firefox and
Blink / Chromium.

It is covered by existing imported WPT test case below:
> html/rendering/bidi-rendering/unicode-bidi-ua-rules.html

Since we haven't aligned with web specification fully, we progress only few more
sub-tests pass for `UA stylesheet rule for unicode-bidi, for <bdo>`.

In 259172@main , WebKit fixed bug to handle bidirectional override and corresponding
glyph content width but it failed to account for `isolate-override`, this fixes it as well.

* Source/WebCore/css/html.css:
(bdo, bdo[dir]):
(bdo): Deleted.
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
(TextUtil::width):
* LayoutTests/fast/css/default-bidi-css-rules-expected.txt:
* LayoutTests/fast/css/default-bidi-css-rules.html:
* LayoutTests/fast/text/international/bidi-LDB-2-HTML.html:

Canonical link: https://commits.webkit.org/287175@main
